Sources/CExecHelpers/exec_helpers.c

```diff
@@ -0,0 +1,90 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include <dirent.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/resource.h>
+#include <limits.h>
+
+#include "include/exec_helpers.h"
+
+// glibc < 2.34 needs this. When running in a container it's possible the userland
+// doesn't have this defined in glibc, but the kernel supports the syscall. It's
+// very trivial to check via invoking it and checking for ENOSYS.
+#ifndef SYS_close_range
+#define SYS_close_range     436
+#endif
+
+static int highest_open_fd_dir(const char *fd_dir) {
+    DIR *dir = opendir(fd_dir);
+    if (dir == NULL) {
+        return -1;
+    }
+
+    int highest_fd = 0;
+    struct dirent *dir_entry = NULL;
+
+    while ((dir_entry = readdir(dir)) != NULL) {
+        if (dir_entry->d_name[0] < '0' || dir_entry->d_name[0] > '9') {
+            continue;
+        }
+
+        int fd = atoi(dir_entry->d_name);
+        if (fd > highest_fd) {
+            highest_fd = fd;
+        }
+     }
+
+    closedir(dir);
+    return highest_fd;
+}
+
+static int get_highest_open_fd() {
+    int highest_fd = -1;
+    #if defined(__APPLE__)
+    highest_fd = highest_open_fd_dir("/dev/fd");
+    #elif defined(__linux__)
+    highest_fd = highest_open_fd_dir("/proc/self/fd");
+    #endif
+
+    if (highest_fd != -1) {
+        return highest_fd;
+    }
+
+    struct rlimit rl;
+    if (getrlimit(RLIMIT_NOFILE, &rl) == 0) {
+        return rl.rlim_cur;
+    }
+
+    // Fallback to sysconf if our pal above didn't work.
+    return sysconf(_SC_OPEN_MAX);
+}
+
+void close_fds_from(unsigned int from) {
```

Comment:
either make this `static` (and put it into the header file) or give it a name space, like `spm_close_fds_from` or so. This is rather likely to clash with other C functions in the global namespace.

Similarly for the other functions in this file.

Reply:
Ack

Reply:
Changed to spm_close_fds_from!